### PR TITLE
Add/history

### PR DESCRIPTION
**Pull request by devceline**

### CODE REVIEW

File: packages/push-client/src/walletClient.ts [hunk 1]
```diff
@@ -232,8 +240,24 @@ export class WalletClient extends IWalletClient {
       account,
       signature,
       (subTopic, subscription) => {
-        if (!subscription) return;
+        if (!subscription) {
+          if (this.core.relayer.subscriber.topics.includes(subTopic)) {
```
Comment (bkrem) [inline packages/push-client/src/walletClient.ts:244]: Can you add a contextual comment here pls regarding the intent of this check and `if` body? 

E.g. why are we checking `relayer.subscriber` and then emptying out `this.messages`?

**Review by bkrem — COMMENTED** [review]
Some tweaks and cleanup needed here but LGTM overall

File: packages/push-client/src/walletClient.ts (hunk 1 above)
Comment (devceline) [reply packages/push-client/src/walletClient.ts:245]: Added some comments and cleared up the flow

Comment (bkrem) [reply packages/push-client/src/walletClient.ts:245]: Ty 🙏

File: packages/push-client/src/controllers/engine.ts [hunk 2]
```diff
@@ -630,10 +634,6 @@ export class PushEngine extends IPushEngine {
       async (event: RelayerTypes.MessageEvent) => {
         const { topic, message, publishedAt } = event;
 
-        if (!message || message.length === 0) {
```
Comment (bkrem) [inline packages/push-client/src/controllers/engine.ts]: Btw I'm prepping a proper upstream fix for this for all clients in `core/controller` here: https://github.com/WalletConnect/walletconnect-monorepo/pull/3137/files

**Review by bkrem — APPROVED** [review]

**Pull request merged**